### PR TITLE
Travis: fix codecov

### PR DESCRIPTION
**Pull request by blueyed**
Trying to fix https://github.com/awesomeWM/awesome/issues/1049

### CODE REVIEW

**Comment by blueyed** [comment]
@psychon 
I assume this to be ready, without knowing why it's needed.
Do  you have a clue?

**Comment by Elv13** [comment]
👍 Thanks for taking care of this

File: tests/run.sh [hunk 1]
```diff
@@ -85,8 +85,11 @@ cd $build_dir
 AWESOME_OPTIONS="$AWESOME_OPTIONS --search lib/"
 # Add test dir (for _runner.lua).
 AWESOME_OPTIONS="$AWESOME_OPTIONS --search $this_dir"
-XDG_CONFIG_HOME="./"
-export XDG_CONFIG_HOME
+export XDG_CONFIG_HOME="./"
+
+# Set LUA_PATH for codecov.
+# (https://github.com/awesomeWM/awesome/issues/1049#issuecomment-247876195)
+export LUA_PATH="$(lua -e 'print(package.path)');lib/?.lua;lib/?/init.lua"
```
Comment (psychon) [inline tests/run.sh]: Unrelated, but I just stumbled upon the following:

> Any ";;" in the value of the environment variable is replaced by the default path. 

https://www.lua.org/manual/5.1/manual.html#pdf-LUA_PATH

I think we should make use of that. :-)

**Comment by psychon** [comment]
Nope, no idea. But `tests/run.sh` only runs awesome, so the difference must be in luacov's "statistics collector" part. However, the idea of this `--search`-flag was to have the same end result in `package.path`.

I have two wishes before this is merged (sorry, I cannot easily test this myself right now):
* Check if the following patch also fixes the problem
* `print(package.path)` at the beginning of rc.lua for a working/non-working build and compare. The important difference *must* be in there.
```diff
diff --git a/tests/run.sh b/tests/run.sh
index b04d62c..1288771 100755
--- a/tests/run.sh
+++ b/tests/run.sh
@@ -82,7 +82,7 @@ fi
 
 cd $build_dir
 
-AWESOME_OPTIONS="$AWESOME_OPTIONS --search lib/"
+AWESOME_OPTIONS="$AWESOME_OPTIONS --search lib"
 # Add test dir (for _runner.lua).
 AWESOME_OPTIONS="$AWESOME_OPTIONS --search $this_dir"
 XDG_CONFIG_HOME="./"
```

**Comment by blueyed** [comment]
@psychon 
Looks like removing the slash fixes it!

**Comment by psychon** [comment]
@blueyed Thanks. Sounds like we should use that as a fix instead. Can you squash the first two commits and remove the third one in this PR?

I guess this means that codecov really fails to see that "lib//awful/util.lua" and "lib/awful/util.lua" is the same file and silently drops coverage information for "lib//awful/util.lua".
CC @stevepeak Just for your information

**Comment by stevepeak** [comment]
@psychon our backend would fix `lib//awful/util.lua` to `lib/awful/util.lua` and merge and duplicate coverage details. Have you discovered otherwise?

**Comment by psychon** [comment]
@stevepeak Yup, looks like that fails for us: https://github.com/awesomeWM/awesome/issues/1049#issuecomment-247928575 (but note that the file names are more complicated than just `/` vs `//`, see the link for working and non-working example)

**Comment by psychon** [comment]
@stevepeak More specifically: No file seems to have `functionaltests`-flag coverage when the upload uses `lib//`, but it works correctly when AFAIK the only difference is turning `lib//` into `lib/`.

**Comment by stevepeak** [comment]
I see talk about `luacov-coveralls` but we are Codecov and do not use this package. Maybe that is the issue?

**Comment by stevepeak** [comment]
I think I found the issue. I'll ping you back here shortly

**Comment by psychon** [comment]
@stevepeak Let's just say "the situation there is complex". ;-) (One travis build handles codecov, another one handles coveralls; together with using different Lua versions etc). But yeah, coveralls is a lot more picky about the path names that it gets and caused me a lot of lost hair to get working.

**Comment by stevepeak** [comment]
Ok. I was wrong. It is not fixing `lib//awful/util.lua` properly. I'll make a server side patch now to fix 👍

**Comment by psychon** [comment]
Thanks a lot!

For this PR: I'm still in favor of removing that one slash (plus the other nice python-"disablation" that Daniel found)

**Comment by blueyed** [comment]
@psychon 
As for this PR, we can still use it?  (it is squashed)

Let's wait / see if codecov handles the extra slash now correctly, before merging it?

**Comment by stevepeak** [comment]
@blueyed deployed that fix yesterday 👍

**Comment by psychon** [comment]
@blueyed :+1: (but feel free to wait a bit for some other PR if you want. Any "random" PR should see a coverage jump from 60% to 70% due to codecov's server side fix.)

**Pull request merged**